src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2744,37 +2744,36 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 				ins->klass = klass;
 			}
 			return ins;
-		}
-		// FIXME: These don't work since Vector2/Vector3 are not handled as SIMD
-#if 0
 		} else if (len == 3 && fsig->param_count == 2 && fsig->params [0]->type == MONO_TYPE_VALUETYPE && fsig->params [1]->type == etype->type) {
 			/* Vector3 (Vector2, float) */
 			int dreg = load_simd_vreg (cfg, cmethod, args [0], NULL);
-			ins = emit_simd_ins (cfg, klass, OP_INSERT_R4, args [1]->dreg, args [2]->dreg);
-			ins->inst_c0 = 2;
+			MonoInst* vec_ins = args [1];
+			if (COMPILE_LLVM (cfg)) {
+				vec_ins = emit_simd_ins (cfg, klass, OP_XWIDEN, args [1]->dreg, -1);
```

Comment:
LGTM, good job!
I noticed that there are `System.Numerics.Vectors` red on linux bionic arm64 but it looks infrastructure related. Possibly re-run of CI could make them green.

Reply:
Checked some jobs before and after changes to vector3. `System.Numerics.Vectors` seems to be red on all of them, along with bunch of different tests on linux bionic arm64, I think it is not related.